services/meta/handler.go

```diff
@@ -79,8 +86,36 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *handler) Close() error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	select {
+	case <-h.closing:
+		// do nothing here
+	default:
+		close(h.closing)
+	}
+	return nil
+}
+
+func (h *handler) isClosed() error {
```

Comment:
This feels like it should return a `bool` and not an error.  It would also make checking it easier instead of doing the `err != nil` dance.